oxide_physics/src/particle.rs

```diff
@@ -0,0 +1,197 @@
+#![allow(dead_code)]
+#![allow(unused_variables)]
+
+use num_traits::pow;
+use oxide_math::commons::vector::*;
+use oxide_math::commons::vector3::Vector3;
+
+extern crate generational_arena;
+use generational_arena::Arena;
+
+pub struct Particle {
+    inverse_mass: f32,
+    damping: f32,
+    position: Vector3,
+    pub velocity: Vector3,
+    force_accum: Vector3,
+    acceleration: Vector3,
+}
+
+impl Particle {
+    fn integrate(&mut self, duration: f32) {
+        // not to integrate things with zero mass
+        if self.inverse_mass <= 0.0f32 {
+            return;
+        }
+        assert!(duration > 0.0);
+
+        // update linear position
+        self.position.x += self.velocity.scale(duration).x;
+        self.position.y += self.velocity.scale(duration).y;
+        self.position.z += self.velocity.scale(duration).z;
+
+        // work out the acceleration from the force
+        let delta = self.force_accum.scale(self.inverse_mass);
+        let resulting_acc = Vector3 {
+            x: self.acceleration.x + delta.x,
+            y: self.acceleration.y + delta.y,
+            z: self.acceleration.z + delta.z,
+        };
+
+        // update linear velocity from the acceleration
+        self.velocity.x += resulting_acc.scale(duration).x;
+        self.velocity.y += resulting_acc.scale(duration).y;
+        self.velocity.z += resulting_acc.scale(duration).z;
+
+        // impose drag
+        self.velocity = self.velocity.scale(pow(self.damping, duration as usize));
+
+        Particle::clear_accumulator(self);
+    }
+
+    fn set_mass(&mut self, mass: f32) {
+        assert!(mass != 0.0f32);
+        self.inverse_mass = (1.0f32) / mass;
+    }
+
+    pub fn get_mass(&self) -> f32 {
+        if self.inverse_mass == 0.0f32 {
+            f32::MAX
+        } else {
+            1.0f32 / self.inverse_mass
+        }
+    }
+
+    fn set_inverse_mass(&mut self, inverse_mass: f32) {
+        self.inverse_mass = inverse_mass;
+    }
+
+    fn get_inverse_mass(&mut self) -> f32 {
+        self.inverse_mass
+    }
+
```

Comment:
I think accessing field direct is better than accessing it by getter and setter. Is there any specific reason that you used getter and setter?

Reply:
Those codes are unnecessary. I will remove the getters and setters not needed.